src/sentry/api/endpoints/project_commits.py

```diff
@@ -22,12 +22,18 @@ def get(self, request: Request, project) -> Response:
                                           commit belongs to.
         :pparam string project_slug: the slug of the project to list the
                                      commits of.
+        :qparam string query: this parameter can be used to create a
+                              "starts with" filter for the commit key.
         """
+        query = request.GET.get("query")
 
         queryset = Commit.objects.filter(
             organization_id=project.organization_id,
             releasecommit__release__releaseproject__project_id=project.id,
-        ).distinct("id")
+        )
+
+        if query:
```

Comment:
If this is a required field, you should return a 400 error like:
https://github.com/getsentry/sentry/blob/001a2108015d28f621bfb8d0831b48f0b85624cb/src/sentry/api/endpoints/project_rules.py#L66-L69

Reply:
I believe `query` is optional in this case